packages/mobile_app/lib/screens/daily/screen/methods/cards.dart

```diff
@@ -8,245 +8,99 @@ extension DailyScreenCardsMethods on _DailyScreenState {
-                          : (sp.cards.swordWasChossen)
-                              ? SmallCard(mode: 1, icon: "sword")
-                              : SmallCard(mode: 0, icon: "sword"),
-                    ),
-                  )
-                : SizedBox(),
-
-            //
-
+            /// here is our small cards row
+            (toShow.moodlet) ? _smallCardBuilder(CardType.TREE) : SizedBox(),
+            (toShow.intuition) ? _smallCardBuilder(CardType.COIN) : SizedBox(),
+            (toShow.luck) ? _smallCardBuilder(CardType.STAR) : SizedBox(),
+            (toShow.ambition) ? _smallCardBuilder(CardType.SWORD) : SizedBox(),
             (toShow.internalStrength)
-                ? Padding(
-                    padding: EdgeInsets.symmetric(horizontal: 1.0),
-                    child: GestureDetector(
-                      onTap: () {
-                        _chooseCup();
-                        _chooseCupBig();
-                      },
-                      child: (sp.cards.cupChoise)
-                          ? SmallCard(mode: 2, icon: "cup")
-                          : (sp.cards.cupWasChossen)
-                              ? SmallCard(mode: 1, icon: "cup")
-                              : SmallCard(mode: 0, icon: "cup"),
-                    ),
-                  )
+                ? _smallCardBuilder(CardType.CUP)
                 : SizedBox(),
-
             //
-
             (toShow.internalStrength == false &&
                     toShow.moodlet == false &&
                     toShow.ambition == false &&
                     toShow.intuition == false &&
                     toShow.luck == false)
-                ? Padding(
-                    padding: EdgeInsets.symmetric(horizontal: 1.0),
-                    child: GestureDetector(
-                      onTap: () {
-                        _chooseStar();
-                        _chooseStarBig();
-                      },
-                      child: (sp.cards.starChoise)
-                          ? SmallCard(mode: 2, icon: "star")
-                          : (sp.cards.starWasChossen)
-                              ? SmallCard(mode: 1, icon: "star")
-                              : SmallCard(mode: 0, icon: "star"),
-                    ),
-                  )
+                ? _smallCardBuilder(CardType.STAR)
                 : SizedBox()
-
-            //
           ],
         ),
       ],
     );
   }
 
-  _chooseTree() {
-    // ignore: invalid_use_of_protected_member
-    setState(() {
-      sp.cards.treeChoise = true;
-      sp.cards.treeWasChossen = true;
-      sp.cards.coinChoise = false;
-      sp.cards.starChoise = false;
-      sp.cards.swordChoise = false;
-      sp.cards.cupChoise = false;
-    });
+  onWatchAdsClick() async {
+    final internet = await internetCheck();
+    if (internet == true) {
+      // ignore: invalid_use_of_protected_member
+      setState(() {
+        final cardAd = getCardAd();
```

Comment:
not quite sure, but mayba a call to `getCardAd();` should happen outside `setState`?